pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -698,7 +752,33 @@ func (r *ReconcileMachinePool) syncMachineSets(
 	for _, ms := range machineSetsToCreate {
 		logger.WithField("machineset", ms.Name).Info("creating machineset")
 		if err := remoteClusterAPIClient.Create(context.Background(), ms); err != nil {
-			logger.WithError(err).Error("unable to create machine set")
+			// Check error message
+			message := "Unable to create MachineSet"
+			if apierrors.IsAlreadyExists(err) {
+				rMS := machineapi.MachineSet{}
+				remoteClusterAPIClient.Get(context.Background(),
+					types.NamespacedName{
+						Name:      ms.Name,
+						Namespace: ms.Namespace,
+					},
+					&machineapi.MachineSet{})
+
+				// FIXME condense these into one?
+				if rMS.Labels[constants.HiveManagedLabel] != "true" {
+					logger.WithError(err).Errorf("%s. A MachineSet with the same name but without a hive managed label already exists."+
+						" To have hive manage this MachineSet, add the label \"hive.openshift.io/managed\" with the value \"true\" to the MachineSet.", message)
+				}
+				if rMS.Labels[machinePoolNameLabel] != pool.Spec.Name {
+					logger.WithError(err).Errorf("%s. A MachineSet with the same name but without a matching MachinePool label already exists."+
+						" To have Machine Pool \"%s\" manage this MachineSet, add the label \"hive.openshift.io/machine-pool\" with the value \"%s\" to the MachineSet.", message, pool.Spec.Name, pool.Spec.Name)
+				}
+				if match, err := matchFailureDomains(ms, rMS, logger); err != nil && !match {
+					logger.WithError(err).Errorf("%s. A MachineSet with the same name but with a different failure domain already exists."+
+						" To have Machine Pool \"%s\" manage this MachineSet, ensure that the failure domain matches", message, pool.Spec.Name)
+				}
+			} else {
+				logger.WithError(err).Error(message)
+			}
```

Comment:
Okay, I get what you're trying to do with this section -- provide useful detail about how we ended up here. But I think it's probably too much effort and too fiddly to be worth it. Dumping the error ought to be sufficient: whoever's debugging ought to be able to go grab and scrutinize the objects themselves.

So I would say just leave the log on L780 and trash the rest.